---
title: "Genomic balance in allopolyploids"
author: "Kevin Bird"
date: "4/26/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(tidyverse)
library(ggplot2)
library(gtools)
library(zoo)
library(cowplot)
library(data.table)
library(readxl)
library(ggpubr)
library(raster)

# FUNCTIONS-----------------------------------------------------------------------------------------
GetSyntelogPairs<- function(R500File,TO1000File){
  TO1000<-read.delim(file=TO1000File, header=TRUE)
  R500<-read.delim(file=R500File, header=TRUE)
  
  colnames(TO1000)<-c("TO1000ID","TO1000Chr","TO1000Strand","TO1000Start","TO1000Stop","TO1000_tName","TO1000Exons","TO1000Length","TO1000GeneID","TO1000GeneName","TO1000Cov","TO1000FPKM","TO1000TPM")
  
  colnames(R500)<-c("R500ID","R500Chr","R500Strand","R500Start","R500Stop","R500_tName","R500Exons","R500Length","R500GeneID","R500GeneName","R500Cov","R500FPKM","R500TPM")
  
  R500  %>%  mutate_if(is.factor,as.character)-> R500
  TO1000  %>%  mutate_if(is.factor,as.character)-> TO1000
  TO1000.R500  %>%  mutate_if(is.factor,as.character)-> TO1000.R500
  
  
    Ortho<-TO1000  %>%  inner_join(TO1000.R500,by="TO1000_tName") %>%  inner_join(R500,by="R500_tName")
  return(Ortho)
  
}

ChiSqTable<-function(ChiSq.test.results){
Sample<-c()
BnC.Obs<-c()
BnA.Obs<-c()
BnC.Exp<-c()
BnA.Exp<-c()
ChiSqStat<-c()
P.value<-c()
for(n in 1:length(ChiSq.test.results)){
  Sample[n]<-names(ChiSq.test.results[n])
  BnC.Obs[n]<-sprintf("%s (%f)",ChiSq.test.results[[n]]$observed[1],(ChiSq.test.results[[n]]$observed[1]/(ChiSq.test.results[[n]]$observed[1]+ChiSq.test.results[[n]]$observed[2])))
  BnA.Obs[n]<-sprintf("%s (%f)",ChiSq.test.results[[n]]$observed[2],(ChiSq.test.results[[n]]$observed[2]/(ChiSq.test.results[[n]]$observed[1]+ChiSq.test.results[[n]]$observed[2])))
  BnC.Exp[n]<-sprintf("%s (%f)",ChiSq.test.results[[n]]$expected[1],(ChiSq.test.results[[n]]$expected[1]/(ChiSq.test.results[[n]]$expected[1]+ChiSq.test.results[[n]]$expected[2])))
  BnA.Exp[n]<-sprintf("%s (%f)",ChiSq.test.results[[n]]$expected[2],(ChiSq.test.results[[n]]$expected[2]/(ChiSq.test.results[[n]]$expected[1]+ChiSq.test.results[[n]]$expected[2])))
  ChiSqStat[n]<-ChiSq.test.results[[n]]$statistic
  P.value[n]<-ChiSq.test.results[[n]]$p.value
}
outTable<-data.frame(Sample=Sample, `BnC with HEs`=BnC.Obs,`BnC without HEs`=BnC.Exp,`BnA with HEs`=BnA.Obs,`BnA without HEs`=BnA.Exp,`Chi Squared`=round(ChiSqStat,2),`P value`=P.value)
return(outTable)
}


colorBlindBlack8  <- c(Nonbiased="#999999", `Dosage Sensistive`="#E69F00", BnC Biased"#56B4E9", `Dosage Insensitive`="#009E73","#F0E442", "#0072B2", "#D55E00", "#CC79A7","#999933")


# READ FILES--------------------------------------------------------------------------------------

TO1000.R500 <- read.delim("TO1000.R500.anchors", header=FALSE, comment.char="#")
colnames(TO1000.R500)<-c("TO1000_tName","R500_tName","BlockSize")

R500SyntelogFiles<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/RNASeq/",pattern="R500",full.names = TRUE)
TO1000SyntelogFiles<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/RNASeq/",pattern="TO1000",full.names = TRUE)

BNapusRNASeqArray<-vector("list", length = length(R500SyntelogFiles))


for (Indiv in 1:length(R500SyntelogFiles)){
  BNapusRNASeqArray[[Indiv]] <-GetSyntelogPairs(R500SyntelogFiles[Indiv],TO1000SyntelogFiles[Indiv])
}

names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5","RS-Parent")

#names<-c("RS-BNapus","RS-Parent")
names(BNapusRNASeqArray)<-names
for(Individ in 1:length(BNapusRNASeqArray)){ 
  BNapusRNASeqArray[[Individ]]<-filter(BNapusRNASeqArray[[Individ]],TO1000Chr %in% c("C1","C2","C3","C4","C5","C6","C7","C8","C9")) 
  BNapusRNASeqArray[[Individ]]<-mutate(BNapusRNASeqArray[[Individ]],SumTPM=TO1000TPM+R500TPM) %>% filter(SumTPM>10) %>% mutate(L2FC=foldchange2logratio(foldchange(num=TO1000TPM+1,denom=R500TPM+1),base=2))  %>%  mutate(Bias= ifelse(L2FC > 3.5,"BnC Biased", ifelse(L2FC < -3.5,"Bna Biased","Nonbiased")))
}


#Get 2:2 Homoeolog list-------------------------------------------------------------------------------------
TwotoTwoHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="TwotoTwo",full.names = T)
TwotoTwoGenes<-c()
for(x in 1:length(TwotoTwoHE)){TwotoTwoGenes[[x]]<-read.delim(file=TwotoTwoHE[x], header=T,sep = ",",stringsAsFactors = F)}
names(TwotoTwoGenes)<-gsub(TwotoTwoHE,pattern = ".csv",replacement = "")
names(TwotoTwoGenes)<-gsub(names(TwotoTwoGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")

# LIST TO OBJECTS--------------------------------------------------------------------------------------


list2env(TwotoTwoGenes,envir =.GlobalEnv)
list2env(BNapusRNASeqArray,envir =.GlobalEnv)

#filter out lowly expressed genes, and get only 2:2 homoeologs identified by WGS
TwotoTwoHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5","RS-Parent")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
  if(grepl("Parent",Individ)){
    TwotoTwoHomoeo[[Individ]]<-get(sprintf("RS-%s",Individ))
    #TwotoTwoHomoeo[[Individ]]<-mutate(TwotoTwoHomoeo[[Individ]],SumTPM=TO1000TPM+R500TPM) %>% filter(SumTPM>10) %>% mutate(L2FC=foldchange2logratio(foldchange(num=TO1000TPM+1,denom=R500TPM+1),base=2))  %>%  mutate(Bias= ifelse(L2FC > 3.5,"BnC Biased", ifelse(L2FC < -3.5,"Bna Biased","Nonbiased")))
    #TwotoTwoHomoeo[[Individ]]$BnC_Density<-rollapply(TwotoTwoHomoeo[[Individ]]$Bias=="BnC Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
    #TwotoTwoHomoeo[[Individ]]$BnA_Density<-rollapply(TwotoTwoHomoeo[[Individ]]$Bias=="Bna Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
    
        }
  else{
    TwotoTwoHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sTwotoTwo",Individ))$TO1000)
    #TwotoTwoHomoeo[[Individ]]<-mutate(TwotoTwoHomoeo[[Individ]],SumTPM=TO1000TPM+R500TPM) %>% filter(SumTPM>10) %>% mutate(L2FC=foldchange2logratio(foldchange(num=TO1000TPM+1,denom=R500TPM+1),base=2))  %>%  mutate(Bias= ifelse(L2FC > 3.5,"BnC Biased", ifelse(L2FC < -3.5,"Bna Biased","Nonbiased")))
    #TwotoTwoHomoeo[[Individ]]$BnC_Density<-rollapply(TwotoTwoHomoeo[[Individ]]$Bias=="BnC Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
    #TwotoTwoHomoeo[[Individ]]$BnA_Density<-rollapply(TwotoTwoHomoeo[[Individ]]$Bias=="Bna Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
    
        }
}
names(TwotoTwoHomoeo)<-gsub(names(TwotoTwoHomoeo),pattern = "^",replacement = "RS-")
```


```{r}

SubgenomeDominanceDFWithHEs<-rbindlist(BNapusRNASeqArray,use.names = T,idcol = "Indiv")
SubgenomeDominanceDFWithoutHEs<-rbindlist(TwotoTwoHomoeo,use.names = T,idcol = "Indiv")

BNapusRNAHEsOnly<-c()
for(Individ in names(BNapusRNASeqArray)){
  BNapusRNAHEsOnly[[Individ]]<-anti_join(BNapusRNASeqArray[[Individ]],TwotoTwoHomoeo[[Individ]],by="TO1000_tName")
}

SubgenomeDominanceDFOnlyHEs<-rbindlist(BNapusRNAHEsOnly,use.names = T,idcol = "Indiv")
```

Dosage analysis game plan: Mick Song and Jeremy Coate provide good framework. Dichotmize dosage sensitivity based on their arabidopsis list of Class I and Class II genes (GO annotation)

Take all HE Genes (anti_join BNapusRNASeqArray, TwotoTwoHomoeo)

Assign them GO terms from At orthologs
-calculate avg dosage response (HE sumTPM/ Parent sumTPM) for GO term
-calculate coefficient of variance (SD/Mean) for GO term


```{r }
#Get Brassica-Arabidopsis orthologs
At.Brassica_Syntelogs <- read.csv("~/Documents/SingleCopyProject/At-Brassica_Syntelogs.csv", stringsAsFactors=FALSE)
#hacky way to deal with the fact that there's a variable number of orthologs and multiple subgenomes in B. oleracea
At.Brassica_Syntelogs<-At.Brassica_Syntelogs  %>%  separate(LF...Bo,sep = ";",into=c("LF_Bo1","LF_Bo2","LF_Bo3","LF_Bo4","LF_Bo5","LF_Bo6"))   %>%  
separate(MF1...Bo,sep = ";",into=c("MF1_Bo1","MF1_Bo2","MF1,Bo3"))  %>% 
separate(MF2...Bo,sep = ";",into=c("MF2_Bo1","MF2_Bo2","MF2_Bo3","MF2_Bo4"))  %>% 
select(-X) %>% 
gather(key=Subgenome,value="Bol_ortho",na.rm = T,3:ncol(.))
  
At.Brassica_Syntelogs$Bol_ortho<-na_if(At.Brassica_Syntelogs$Bol_ortho,"-")  %>%  na_if(.,"na ") %>% na_if(.,"no At orthologs in Brassicas  ")
At.Brassica_Syntelogs<-drop_na(At.Brassica_Syntelogs)


#GO ANNOTATION###################################################################
#Get GO annotation of Arabidopsis genes from Song et al. 2020 supplemental
At_GO <- read_excel("~/Downloads/TPC2019-RA-00832R2_Supplemental_Dataset_1.xlsx", sheet = "GO raw data", skip = 2)
At_GO<-At_GO[1:3]
At_GO<-At_GO %>% rename(TAIR10=`Gene`)

#Get Duplicate Dosage sensitivity class based on GO category from Song et al. 2020 supplemental
At_DosageSensitivity_FromGO <- read_excel("~/Downloads/TPC2019-RA-00832R2_Supplemental_Dataset_1.xlsx", sheet = "GO PRV DR - C24", skip = 1)
At_DosageSensitivity_FromGO<-At_DosageSensitivity_FromGO[c(1,3)]
At_DosageSensitivity_FromGO<-At_DosageSensitivity_FromGO %>% rename(GO=`GO Term`)

#Start joining tables
BrassicaOtrho_DosageClassfromGO<-inner_join(At.Brassica_Syntelogs,At_GO,by="TAIR10") %>%  inner_join(.,y=At_DosageSensitivity_FromGO,by="GO")  %>%  drop_na() %>% rename(TO1000GeneID=Bol_ortho)

#Combine with Parental expression to get the HE response variance
SubgenomeDominance_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFOnlyHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")

TwoTwo_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFWithoutHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")
  
```

Plot polyploid Response Variance based on GO categories
```{r PRV,echo=T}

#Plot average for dosage classes across all lines

TwoTwo_Main_Raw<- TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% group_by(GO,Class) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(avg_exp_response,sd_exp_response,PRV)  %>% mutate(count=n()) %>% distinct() %>% filter(count>20) 

TwoTwo_Main_Plot<- TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% group_by(GO,Class) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(PRV)  %>% mutate(count=n()) %>% distinct() %>% filter(count>20) 

TwoTwo_Main_Plot %>% ggplot(aes(y=PRV,x=Class,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(outlier.shape = NA,alpha=0.75,width=0.5)+ylab("Polyploid Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)

#Plot for Subgenome bias and dosage classes across all lines
TwoTwo_HEB_Raw <- TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% group_by(GO,Class,Bias.y) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(avg_exp_response,sd_exp_response,PRV)  %>% mutate(count=n()) %>% distinct() 


TwoTwo_HEB_Plot <- TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% group_by(GO,Class,Bias.y) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(PRV)  %>% mutate(count=n()) %>% distinct() %>% filter(count>20) 

TwoTwo_HEB_Plot %>% ggplot(aes(y=PRV,x=Bias.y,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(outlier.shape = NA,alpha=0.75)+ylab("Polyploid Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)

#Plot by generation

TwoTwo_Gen_Raw<- TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% separate(Indiv,into=c("Line","Generation"),sep="00") %>% group_by(GO,Class,Generation) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(avg_exp_response,sd_exp_response,PRV)   %>% mutate(count=n()) %>% distinct() %>% filter(count>20)
 
TwoTwo_Gen_Plot<-TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% separate(Indiv,into=c("Line","Generation"),sep="00") %>% group_by(GO,Class,Generation) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y/2))) %>% mutate(avg_exp_response= mean(SumTPM.x/(SumTPM.y/2))) %>% mutate(PRV=sd_exp_response/avg_exp_response) %>% summarise(PRV)  %>% mutate(count=n()) %>% distinct() %>% filter(count>20) 

TwoTwo_Gen_Plot%>% ggplot(aes(y=PRV,x=Generation,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(outlier.shape = NA,alpha=0.75)+ylab("Polyploid Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)+scale_x_discrete(limits=c("S1","S5","S10"))

```
```{r Analysis of HERV}
#Plot Homoeologous Exchange Response Variance based on GO categories

BNapusRNAHEsOnly<-c()
for(Individ in names(BNapusRNASeqArray)){
  BNapusRNAHEsOnly[[Individ]]<-anti_join(BNapusRNASeqArray[[Individ]],TwotoTwoHomoeo[[Individ]],by="TO1000_tName")
}
SubgenomeDominanceDFOnlyHEs<-rbindlist(BNapusRNAHEsOnly,use.names = T,idcol = "Indiv")


SubgenomeDominance_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFOnlyHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")


#Plot average for dosage classes across all lines

SubgenomeDominance_w_DosageClassFromGO %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class) %>% mutate(HERV=sd(SumTPM.x/SumTPM.y)/mean(SumTPM.x/SumTPM.y)) %>% summarise(HERV) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% ggplot(aes(y=HERV,x=Class,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(width=0.5,outlier.shape = NA,alpha=0.75)+ylab("Homoeologous Exchange Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)

#Plot for Subgenome bias and dosage classes across all lines

SubgenomeDominance_w_DosageClassFromGO %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class,Bias.y) %>% mutate(HERV=sd(SumTPM.x/SumTPM.y)/mean(SumTPM.x/SumTPM.y)) %>% summarise(HERV) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% ggplot(aes(y=HERV,x=Bias.y,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(outlier.shape = NA,alpha=0.75)+ylab("Homoeologous Exchange Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)
#Plot by generation

SubgenomeDominance_w_DosageClassFromGO %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% separate(Indiv,into=c("Line","Generation"),sep="00") %>% group_by(GO,Class,Generation) %>% mutate(HERV=sd(SumTPM.x/SumTPM.y)/mean(SumTPM.x/SumTPM.y)) %>% summarise(HERV) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% ggplot(aes(y=HERV,x=Generation,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = .25))+geom_boxplot(outlier.shape = NA,alpha=0.75)+ylab("Homoeologous Exchange Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)+scale_x_discrete(limits=c("S1","S5","S10"))
```

# Calculate expression response to different kinds of homoeologous exchange events

```{r}

#ZeroToFour
ZerotoFourHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="ZerotoFour",full.names = T,)
ZerotoFourGenes<-c()


ZerotoFourGenes<-lapply(ZerotoFourHE, function(x) {
    if (!file.size(x) == 4) {
        read.delim(x, header=T,sep = ",",stringsAsFactors = F)
    }
})

names(ZerotoFourGenes)<-gsub(ZerotoFourHE,pattern = ".csv",replacement = "")
names(ZerotoFourGenes)<-gsub(names(ZerotoFourGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")


list2env(ZerotoFourGenes,envir =.GlobalEnv)

ZerotoFourHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
    ZerotoFourHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sZerotoFour",Individ))$TO1000)
}
names(ZerotoFourHomoeo)<-gsub(names(ZerotoFourHomoeo),pattern = "^",replacement = "RS-")


SubgenomeDominanceDFZerotoFourHEs<-rbindlist(ZerotoFourHomoeo,use.names = T,idcol = "Indiv")
ZerotoFour_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFZerotoFourHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")


#OnetoThree
OnetoThreeHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="OnetoThree",full.names = T,)
OnetoThreeGenes<-c()


OnetoThreeGenes<-lapply(OnetoThreeHE, function(x) {
    if (!file.size(x) == 4) {
        read.delim(x, header=T,sep = ",",stringsAsFactors = F)
    }
})

names(OnetoThreeGenes)<-gsub(OnetoThreeHE,pattern = ".csv",replacement = "")
names(OnetoThreeGenes)<-gsub(names(OnetoThreeGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")


list2env(OnetoThreeGenes,envir =.GlobalEnv)

OnetoThreeHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
    OnetoThreeHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sOnetoThree",Individ))$TO1000)
}
names(OnetoThreeHomoeo)<-gsub(names(OnetoThreeHomoeo),pattern = "^",replacement = "RS-")


SubgenomeDominanceDFOnetoThreeHEs<-rbindlist(OnetoThreeHomoeo,use.names = T,idcol = "Indiv")
OnetoThree_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFOnetoThreeHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")



#OnetoThree
OnetoThreeHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="OnetoThree",full.names = T,)
OnetoThreeGenes<-c()


OnetoThreeGenes<-lapply(OnetoThreeHE, function(x) {
    if (!file.size(x) == 4) {
        read.delim(x, header=T,sep = ",",stringsAsFactors = F)
    }
})

names(OnetoThreeGenes)<-gsub(OnetoThreeHE,pattern = ".csv",replacement = "")
names(OnetoThreeGenes)<-gsub(names(OnetoThreeGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")


list2env(OnetoThreeGenes,envir =.GlobalEnv)

OnetoThreeHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
    OnetoThreeHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sOnetoThree",Individ))$TO1000)
}

names(OnetoThreeHomoeo)<-gsub(names(OnetoThreeHomoeo),pattern = "^",replacement = "RS-")


SubgenomeDominanceDFOnetoThreeHEs<-rbindlist(OnetoThreeHomoeo,use.names = T,idcol = "Indiv")
OnetoThree_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFOnetoThreeHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")

#ThreetoOneHE

ThreetoOneHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="ThreetoOne",full.names = T,)
ThreetoOneGenes<-c()


ThreetoOneGenes<-lapply(ThreetoOneHE, function(x) {
    if (!file.size(x) == 4) {
        read.delim(x, header=T,sep = ",",stringsAsFactors = F)
    }
})

names(ThreetoOneGenes)<-gsub(ThreetoOneHE,pattern = ".csv",replacement = "")
names(ThreetoOneGenes)<-gsub(names(ThreetoOneGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")


list2env(ThreetoOneGenes,envir =.GlobalEnv)

ThreetoOneHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
    ThreetoOneHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sThreetoOne",Individ))$TO1000)
}
names(ThreetoOneHomoeo)<-gsub(names(ThreetoOneHomoeo),pattern = "^",replacement = "RS-")


SubgenomeDominanceDFThreetoOneHEs<-rbindlist(ThreetoOneHomoeo,use.names = T,idcol = "Indiv")
ThreetoOne_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFThreetoOneHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")


#FourtoZero

FourtoZeroHE<-list.files(path="~/Documents/Birdetal2020Data_and_Figures/WGS",pattern="FourtoZero",full.names = T,)
FourtoZeroGenes<-c()


FourtoZeroGenes<-lapply(FourtoZeroHE, function(x) {
    if (!file.size(x) == 4) {
        read.delim(x, header=T,sep = ",",stringsAsFactors = F)
    }
})

names(FourtoZeroGenes)<-gsub(FourtoZeroHE,pattern = ".csv",replacement = "")
names(FourtoZeroGenes)<-gsub(names(FourtoZeroGenes),pattern = "/home/bird/Documents/Birdetal2020Data_and_Figures/WGS/",replacement = "")


list2env(FourtoZeroGenes,envir =.GlobalEnv)

FourtoZeroHomoeo<-c()
names<- c("RS-100S1","RS-100S10","RS-100S5","RS-1100S1","RS-1100S10","RS-1100S5","RS-200S1","RS-200S10","RS-200S5","RS-300S1","RS-300S10","RS-400S1","RS-400S5","RS-600S1","RS-600S10","RS-600S5")

IndivNames<-mixedsort(gsub(names,pattern = "RS-",replacement = ""))

for(Individ in IndivNames){ 
    FourtoZeroHomoeo[[Individ]]<-subset(get(sprintf("RS-%s",Individ)),TO1000_tName %in% get(sprintf("Ds-%sFourtoZero",Individ))$TO1000)
    #FourtoZeroHomoeo[[Individ]]<-mutate(FourtoZeroHomoeo[[Individ]],SumTPM=TO1000TPM+R500TPM) %>% filter(SumTPM>10) %>% mutate(L2FC=foldchange2logratio(foldchange(num=TO1000TPM+1,denom=R500TPM+1),base=2))  %>%  mutate(Bias= ifelse(L2FC > 3.5,"BnC Biased", ifelse(L2FC < -3.5,"Bna Biased","Nonbiased")))
    #FourtoZeroHomoeo[[Individ]]$BnC_Density<-rollapply(FourtoZeroHomoeo[[Individ]]$Bias=="BnC Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
    #FourtoZeroHomoeo[[Individ]]$BnA_Density<-rollapply(FourtoZeroHomoeo[[Individ]]$Bias=="Bna Biased", width = 10, by = 1, FUN = sum, na.rm = TRUE,partial=T)
}

names(FourtoZeroHomoeo)<-gsub(names(FourtoZeroHomoeo),pattern = "^",replacement = "RS-")


SubgenomeDominanceDFFourtoZeroHEs<-rbindlist(FourtoZeroHomoeo,use.names = T,idcol = "Indiv")
FourtoZero_w_DosageClassFromGO<-inner_join(SubgenomeDominanceDFFourtoZeroHEs,BrassicaOtrho_DosageClassfromGO,by="TO1000GeneID") %>% inner_join(y=BNapusRNASeqArray$`RS-Parent`,by="TO1000_tName")


```

#analyze HERV for each dosage category seperately so all genes have same expected expression response against the 2:2 expression level
```{r}

#Total Comparison (Consider changing to analyzing each for against 2:2 so at least we're not mixing expectations)
ZerotoFour_HERV<-ZerotoFour_w_DosageClassFromGO  %>% filter(R500TPM.y > 0) %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class) %>%  mutate(avg_exp_response=mean(SumTPM.x/SumTPM.y)) %>% mutate(sd_exp_response=sd(SumTPM.x/SumTPM.y)) %>% mutate(HERV=sd_exp_response/avg_exp_response) %>% mutate(dosage="0:4")

OnetoThree_HERV<-OnetoThree_w_DosageClassFromGO  %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class) %>% mutate(avg_exp_response=mean(SumTPM.x/SumTPM.y)) %>% mutate(sd_exp_response=sd(SumTPM.x/SumTPM.y)) %>% mutate(HERV=sd_exp_response/avg_exp_response)  %>% mutate(dosage="1:3")

ThreetoOne_HERV<-ThreetoOne_w_DosageClassFromGO %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class) %>% mutate(avg_exp_response=mean(SumTPM.x/(SumTPM.y))) %>% mutate(sd_exp_response=sd(SumTPM.x/(SumTPM.y))) %>% mutate(HERV=sd_exp_response/avg_exp_response)  %>% mutate(dosage="3:1")

FourtoZero_HERV<-FourtoZero_w_DosageClassFromGO %>% filter(TO1000TPM.y > 0) %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(GO,Class) %>%  mutate(avg_exp_response=mean(SumTPM.x/SumTPM.y)) %>% mutate(sd_exp_response=sd(SumTPM.x/SumTPM.y)) %>% mutate(HERV=sd_exp_response/avg_exp_response)  %>% mutate(dosage="4:0") 

ReanalyHERVMain<-rbind(ZerotoFour_HERV,OnetoThree_HERV,ThreetoOne_HERV,FourtoZero_HERV) 

HERV_Main_separate <-ReanalyHERVMain %>% group_by(GO,Class,dosage) %>% drop_na(HERV) %>% summarise(avg_exp_response,sd_exp_response,HERV,dosage) %>% mutate(count=n()) %>% distinct() 

HERVMainPlot<- ReanalyHERVMain %>% drop_na(HERV) %>% summarise(avg_exp_response,sd_exp_response,HERV,dosage) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% summarise(avgHERV=mean(HERV),count=mean(count))

HERVMainPlot %>% ggplot(aes(y=avgHERV,x=Class,fill=Class))+geom_point(position=position_jitterdodge(dodge.width = .75,jitter.width = 0.5))+geom_boxplot(width=0.5,outlier.shape = NA,alpha=0.75)+ylab("Homoeologous Exchange Response Variance")+theme_cowplot(12)+scale_fill_discrete(label=c("Dosage \n Insensitive","Dosage \n Sensitive"))+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)

```

#Compare Polyploid dosage response to HE dosage response to see if they are distinct

```{r compare HERV and PRV}

HERV<-ReanalyHERVMain %>% group_by(GO,Class,dosage) %>% drop_na(HERV) %>% summarise(Response_Variance=HERV) %>% mutate(count=n()) %>% distinct() %>% filter(count>20)

PRV<-TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% mutate(dosage="2:2") %>% group_by(GO,Class,dosage) %>% mutate(PRV=sd(SumTPM.x/SumTPM.y)/mean(SumTPM.x/SumTPM.y)) %>% summarise(Response_Variance=PRV)  %>% mutate(count=n()) %>% distinct() %>% filter(count>20)

Plot_test<-rbind(HERV,PRV)

ggplot(data=Plot_test,aes(x=dosage,y=Response_Variance,group=dosage,fill=dosage))+geom_jitter()+geom_boxplot(outlier.shape = NA,alpha=0.75)+theme_cowplot(12)+stat_compare_means(comparisons=list(c("2:2", "0:4"),c("2:2","1:3"),c("2:2","3:1"),c("2:2","4:0")),label = "p.signif",size=4.5,label.x = 2)+ylab("Coeficient of Variance")+facet_wrap(~Class)

#Check Exp Lvl by GO term Class

#Plot of Average TPM of homoeolog pairs in a GO term w/ > 20 genes comparing Class I and Class II avg expression for PRV dataset (2:2 homoeologs)
TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent") %>% group_by(GO,Class) %>% summarise(SumTPM.x) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% summarise(avgSumTPM=mean(SumTPM.x)) %>% ggplot(data=.,aes(x=Class,y=avgSumTPM,fill=Class))+geom_boxplot(outlier.shape = NA ,alpha=0.5)+ylim(0,400)+theme_cowplot(12)+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)

#Plot of Average TPM of homoeolog pairs in a GO term w/ > 20 genes comparing Class I and Class II avg expression for HERV dataset (HEs excluding likely aneuploids)

ReanalyHERVMain %>% group_by(GO,Class) %>% summarise(SumTPM.x) %>% mutate(count=n()) %>% distinct() %>% filter(count>20) %>% summarise(avgSumTPM=mean(SumTPM.x)) %>% ggplot(data=.,aes(x=Class,y=avgSumTPM,fill=Class))+geom_boxplot(outlier.shape = NA,alpha=0.5)+ylim(0,400)+theme_cowplot(12)+stat_compare_means(method = "kruskal.test",label="p.format",size=4.5,label.x = 2)


#Check proportional breakdown of GO term Classes

BalancedClassBrkdwn<-table(TwoTwo_w_DosageClassFromGO %>% filter(Indiv != "RS-Parent")%>%group_by(Indiv)%>%summarise(Class))
HEDosageClassBrkdwn<-table(ReanalyHERVMain %>% filter(!(Indiv == "RS-1100S10" & TO1000Chr.x =="C4")) %>%filter(TO1000Chr.x != "C1" & TO1000Chr.x !="C2" & R500Chr.x !="A10") %>% group_by(Indiv) %>% summarise(Class))

```
